src/config_cache.c

```diff
@@ -86,7 +86,8 @@ int git_config__cvar(int *out, git_config *config, git_cvar_cached cvar)
 	struct map_data *data = &_cvar_maps[(int)cvar];
 	git_config_entry *entry;
 
-	git_config__lookup_entry(&entry, config, data->cvar_name, false);
+	if ((error = git_config__lookup_entry(&entry, config, data->cvar_name, false)) < 0)
+		return error;
```

Comment:
This bypasses setting our output variable to the default value. We were doing the equivalent of detecting a not-found entry (or whatever error) by checking `entry` against `NULL` in the check below.

This indicates we don't have nearly enough testing here, since now we're not setting `*out` anymore, but the tests still pass.